src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/SocketsHttpHandler.cs

```diff
@@ -531,8 +531,11 @@ private HttpMessageHandlerStage SetupHandlerChain()
 
             // MetricsHandler should be descendant of DiagnosticsHandler in the handler chain to make sure the 'http.request.duration'
             // metric is recorded before stopping the request Activity. This is needed to make sure that our telemetry supports Exemplars.
-            handler = new MetricsHandler(handler, settings._meterFactory, out Meter meter);
-            settings._metrics = new SocketsHttpHandlerMetrics(meter);
+            if (MetricsHandler.IsGloballyEnabled())
+            {
+                handler = new MetricsHandler(handler, settings._meterFactory, out Meter meter);
+                settings._metrics = new SocketsHttpHandlerMetrics(meter);
```

Comment:
We'd have to update other places that rely on `_metrics` being set
https://grep.app/search?f.repo=dotnet%2Fruntime&f.repo.pattern=runtime&q=._metrics

Do we have a need for the opt-out switch outside of browser scenarios?

Reply:
> Do we have a need for the opt-out switch outside of browser scenarios?

Maybe we don't need the opt-out for functional reasons.
But in browser, most of the time, this is just dead weight and I wonder if that should be opt-in for the browser.


> We'd have to update other places that rely on `_metrics` being set.

Thanks, I should probably cover that by some unit test. Or maybe I could split this HTTP into separated PR.

Reply:
Moved the change to https://github.com/dotnet/runtime/pull/113528 and will incorporate your feedback there later.